File: LeetCodeProblemsCMake/Problems/LeetCode594LongestHarmoniousSubsequence.cpp
//
// Created by saeed on 6/30/25.
//

#include "LeetCode594LongestHarmoniousSubsequence.h"

#include <unordered_map>
using namespace std;

int LeetCode594LongestHarmoniousSubsequence::findLHS(vector<int> &nums) {
    unordered_map<int, int> map{};
    for (auto num: nums) {
        map[num]++;
    }
    int result{0};
    int current{0};
    for (auto &pair: map) {
        current = pair.second;
        if (!map.contains(pair.first + 1))
            continue;
        current += map[pair.first + 1];
        if (current > result) {
            result = current;
        }
    }
    return result;
}

File: LeetCodeProblemsGo/LeetCode594LongestHarmoniousSubsequence/LeetCode594LongestHarmoniousSubsequence.go
package LeetCode594LongestHarmoniousSubsequence

func findLHS(nums []int) int {
	numsMap := make(map[int]int)
	for _, num := range nums {
		numsMap[num]++
	}
	current := 0
	max := 0
	for key, value := range numsMap {
		current = value
		if _, exists := numsMap[key+1]; !exists {
			continue
		}
		current += numsMap[key+1]
		if current > max {
			max = current
		}
	}
	return max
}
